pyquil/reference_simulator.py

```diff
@@ -165,6 +165,25 @@ def __init__(self, n_qubits: int, rs: RandomState = None):
         self.rs = rs
         self.density = np.zeros((2 ** n_qubits, 2 ** n_qubits), dtype=np.complex128)
         self.density[0, 0] = complex(1.0, 0)
```

Comment:
Again this is code repeated in `self.reset()` -- you might instead just call `self.reset()` here.

Reply:
I'm pretty new to all this ... so I have a question. Is that an okay (or a standard thing) to do in `__init__`?

Reply:
Yes. For example, in the qvm, the initialisation functions all call something like `bring-to-zero-state`. The benefit being that you don't duplicate code. What if your definition of "zero state" changed, and it was actually now all zeros with a one in the last position? You would have to change multiple parts of the code to reflect this change, which means you're more likely to miss something and break stuff.

Reply:
- [x] todo